siddhi-extensions/nlp/src/main/java/org/wso2/siddhi/gpl/extension/nlp/NameEntityTypeStreamProcessor.java

```diff
@@ -118,56 +116,61 @@ private void initPipeline() {
     }
 
     @Override
-    protected void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processor processor, StreamEventCloner streamEventCloner, ComplexEventPopulater complexEventPopulater) {
-        while (streamEventChunk.hasNext()) {
-            StreamEvent event = streamEventChunk.next();
-
-            if (logger.isDebugEnabled()) {
-                logger.debug(String.format("Event received. Entity Type:%s GroupSuccessiveEntities:%s " +
-                        "Event:%s", entityType.name(), groupSuccessiveEntities, event));
-            }
-
-            Annotation document = new Annotation(attributeExpressionExecutors[2].execute(event).toString());
-            pipeline.annotate(document);
-
-            StreamEvent newEvent = null;
-            List<CoreMap> sentences = document.get(CoreAnnotations.SentencesAnnotation.class);
-
-            if (groupSuccessiveEntities) {
-                String word;
-                boolean added = false;
+    protected void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processor nextProcessor, StreamEventCloner streamEventCloner, ComplexEventPopulater complexEventPopulater) {
+        synchronized (this) {
+            while (streamEventChunk.hasNext()) {
+                StreamEvent event = streamEventChunk.next();
+
+                if (logger.isDebugEnabled()) {
+                    logger.debug(String.format("Event received. Entity Type:%s GroupSuccessiveEntities:%s " +
+                            "Event:%s", entityType.name(), groupSuccessiveEntities, event));
+                }
 
-                for (CoreMap sentence : sentences) {
-                    for (CoreLabel token : sentence.get(CoreAnnotations.TokensAnnotation.class)) {
-                        if (entityType.name().equals(token.get(CoreAnnotations.NamedEntityTagAnnotation.class))) {
-                            word = token.get(CoreAnnotations.TextAnnotation.class);
-                            if (added) {
-                                word = newEvent.getAttribute(inStreamParamPosition).toString() + " " + word;
-                                complexEventPopulater.populateComplexEvent(newEvent, new Object[]{word});
+                Annotation document = new Annotation(attributeExpressionExecutors[2].execute(event).toString());
+                pipeline.annotate(document);
+
+                StreamEvent newEvent = null;
+                List<CoreMap> sentences = document.get(CoreAnnotations.SentencesAnnotation.class);
+
+                if (groupSuccessiveEntities) {
+                    String word;
+                    String matchedWord = null;
+                    boolean added = false;
```

Comment:
I would find this variable name more meaningful and readable if it had a "has" prefix or similar.